x/evm/keeper/vote_handler.go

```diff
@@ -0,0 +1,150 @@
+package keeper
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/x/evm/types"
+	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	vote "github.com/axelarnetwork/axelar-core/x/vote/exported"
+)
+
+// NewVoteHandler returns the handler for processing vote delivered by the vote module
+func NewVoteHandler(cdc codec.Codec, keeper types.BaseKeeper, nexus types.Nexus) vote.VoteHandler {
+	return func(ctx sdk.Context, chainName string, result *vote.Vote) error {
+
+		chain, ok := nexus.GetChain(ctx, chainName)
+		if !ok {
+			return fmt.Errorf("%s is not a registered chain", chainName)
+		}
+
+		if err := validateChainActivated(ctx, nexus, chain); err != nil {
+			return err
+		}
+
+		events, err := types.UnpackEvents(cdc, result.Results)
+		if err != nil {
+			return err
+		}
+		var errors []error
+		for _, event := range events {
+			var err error
+
+			// validate event
+			err = event.Validate()
+			if err != nil {
+				errors = append(errors, fmt.Errorf("event %d: %s", event.Index, err.Error()))
+				continue
+			}
+
+			// check if event confirmed before
+			eventID := event.GetID()
+			if _, ok := keeper.ForChain(chainName).GetEvent(ctx, eventID); ok {
+				errors = append(errors, fmt.Errorf("event %s is already confirmed", eventID))
+				continue
+			}
+
+			switch e := event.GetEvent().(type) {
+			case *types.Event_Transfer:
+				err = handleVoteConfirmDeposit(ctx, keeper, nexus, chain, e)
+			case *types.Event_TokenDeployed:
+				err = handleVoteConfirmToken(ctx, keeper, chain, e)
+			default:
+				err = fmt.Errorf("event %d: unsupported event type %T", event.Index, event)
+			}
+
+			if err != nil {
+				errors = append(errors, fmt.Errorf("event %d: %s", event.Index, err.Error()))
+				continue
+			}
+
+			keeper.ForChain(chain.Name).SetConfirmedEvent(ctx, event)
+		}
+
+		if len(errors) != 0 {
```

Comment:
Why do you want to collect all the errors and log them together? Also, I'd prefer them to be logged individually with a loop to avoid long lines.

Reply:
these errors also returned to the tx res as log
```
&types.VoteResponse{Log: fmt.Sprintf("failed to process %s", err.Error())}
```